### PR TITLE
BSA-6 | Main menu, button to start game, also display highscore

### PR DESCRIPTION
**Pull request by danmcgill6**
[BSA-6](https://trello.com/c/IQRyOhDy/6-main-menu-button-to-start-game-also-display-highscore)

**Changes:**

Refactored the entire code base.

### CODE REVIEW

**Review by ohmlnz — CHANGES_REQUESTED** [review]
Great job implementing both local storage and navigation into the game! I've left a bunch of comments, most of them being cosmetic stuff. 

Two things I wasn't able to leave comments for though:

1. `.DS_Store` should be removed, it's a metadata file specific to macos that contains info about the folder (icon size, background color, etc.). I've added it to our `.gitignore` in one of my previous stories, you should be able to see it if your branch has been rebased
2. I think by convention everything that gets compiled should be within `src`. I'm having a hard time following the new structure you implemented. We can talk about it in person on Tuesday.

File: engine/src/game.js [hunk 1]
```diff
@@ -88,19 +95,49 @@ export default class Game extends PureComponent {
...
+            <Button
+              onPress={() => {
+                this.setState({
+                  running: !this.state.running,
+                  isPaused: !this.state.isPaused,
+                });
+              }}
+              title="Pause"
+              color="#841584"
+              accessibilityLabel="Start avoiding shit"
```
[the remark this answers is not in the document]
Comment (danmcgill6) [reply engine/src/game.js:137]: I'm actually not sure what this label does, whats displayed is the title

File: engine/utils/localStorage.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+export const getData = async (storageKey: string): Promise<any> => {
+  try {
+    const value = await AsyncStorage.getItem(storageKey);
+    if (value !== null) {
+      return value;
+    }
+  } catch (e) {
+    console.log("error", e);
```
[the remark this answers is not in the document]
Comment (ohmlnz) [reply engine/utils/localStorage.ts:21]: [https://trello.com/c/ll2TK2PV/35-implement-proper-error-handling-to-logging-service](https://trello.com/c/ll2TK2PV/35-implement-proper-error-handling-to-logging-service)

File: engine/src/game.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ohmlnz) [reply engine/src/game.js:137]: > Great job implementing both local storage and navigation into the game! I've left a bunch of comments, most of them being cosmetic stuff.
> 
> Two things I wasn't able to leave comments for though:
> 
> 1. `.DS_Store` should be removed, it's a metadata file specific to macos that contains info about the folder (icon size, background color, etc.). I've added it to our `.gitignore` in one of my previous stories, you should be able to see it if your branch has been rebased
> 2. I think by convention everything that gets compiled should be within `src`. I'm having a hard time following the new structure you implemented. We can talk about it in person on Tuesday.

[https://trello.com/c/sGOiRWZz/36-explore-refactoring-file-structure](https://trello.com/c/sGOiRWZz/36-explore-refactoring-file-structure)

File: engine/screens/MainMenu/styles.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { StyleSheet } from "react-native";
+
+export const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    alignItems: "center",
+    justifyContent: "center",
+    padding: 24,
+    backgroundColor: "#fff",
```
[the remark this answers is not in the document]
Comment (ohmlnz) [reply engine/screens/MainMenu/styles.ts:9]: [https://trello.com/c/Ruyuoxwq/37-homogenize-color-codes-to-one-value-name-rgba-hex-or-shorthand-hex-and-add-css-variables](https://trello.com/c/Ruyuoxwq/37-homogenize-color-codes-to-one-value-name-rgba-hex-or-shorthand-hex-and-add-css-variables)

File: App.js [hunk 4]
```diff
@@ -27,6 +28,10 @@ export default class App extends Component {
       );
     }
 
-    return <Engine />;
+    return (
+      <NavigationContainer>
+        <Navigator />
```
[the remark this answers is not in the document]
Comment (ohmlnz) [reply App.js:33]: [https://trello.com/c/sGOiRWZz/36-explore-refactoring-file-structure](https://trello.com/c/sGOiRWZz/36-explore-refactoring-file-structure)

**Review by ohmlnz — APPROVED** [review]

**Comment by ohmlnz** [comment]
🔥🔥🔥

**Pull request merged**